### PR TITLE
feat: Eager prebid ab test

### PR DESCRIPTION
**Pull request by Jakeii**
## What does this change?
Introduce a test to run prebid ASAP on page load rather than when we want the ad to load, this will be a multi-variant test, testing different variations of the lazy-load margin less than and equal to the current value of 20% of the viewport as the load time of ads should be reduced, specifically `20%`, `17.5%`, `15%`, `12.5%` and `10%`.

- Created a `request-bids` file that contains the logic for requesting bids and refreshing bids from a9 and prebid.
- The requestBids functions for a9 and prebid have been refactored to accept multiple `Advert` instances
- addSlot now returns a promise with the `Advert` inserted, so that they can be collected and prebid can be run for these slots when space-finder is finished in article-body-adverts and liveblog-adverts.
- bids are now requested for static slots in fill-advert-slots
- The prebid promise is "remembered" on the `Advert` class as there's a case when an ad is in view and loaded before space-finder finishes.

## Why?
At the moment prebid is triggered when we want to load the ad, at the moment this is 20% before the ad slot is in view:
image

After introducing this change it will be more like:
image

This should in theory speed up the loading of ads as they are scrolled into view, and hopefully have no to little effect on the loading of ads already in view, or the performance of page loads in general.

More details in this doc: https://docs.google.com/document/d/14DGHvLAcrmEucunybSQP43mW7G_mDyNNlnBNQDTOhtw/edit#heading=h.garspllpczku

### CODE REVIEW

File: bundle/src/projects/common/modules/experiments/tests/eager-prebid.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	expiry: '2023-03-30',
+	audience: 5 / 100,
+	audienceOffset: 35 / 100,
+	audienceCriteria: 'All pageviews',
+	successMeasure:
+		'Ads lazy load faster, without affecting the page load time',
+	description: 'Test the impact of running prebid on page load',
+	variants: [
+		{ id: 'control', test: bypassMetricsSampling },
+		{ id: 'variant-20', test: bypassMetricsSampling },
```
Comment (domlander) [inline bundle/src/projects/common/modules/experiments/tests/eager-prebid.ts:17]: This one's just floating an idea and completely up to you, but I'd probably name these: `variant-200`, `variant-175`, `variant-150`, `variant-125`, `variant-100`

**Review by domlander — APPROVED** [review]

File: bundle/src/projects/commercial/modules/header-bidding/request-bids.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	return [
+		{
+			...hbSlot,
+			sizes: [[advertSize[0], advertSize[1]] as AdSize],
+		},
+	];
+};
+
+/**
+ * This is used by to request bids for a single advert. This should only be called if an ad is already in the viewport and load-advert is invoked immediately, before space-finder is finished and prebid is called for all dynamic slots.
```
[the remark this answers is not in the document]
Comment (Jakeii) [reply bundle/src/projects/commercial/modules/header-bidding/request-bids.ts:35]: thanks!

File: bundle/src/projects/commercial/modules/dfp/load-advert.ts [hunk 3]
```diff
@@ -46,11 +19,11 @@ export const loadAdvert = (advert: Advert): void => {
 			// The display needs to be called, even in the event of an error.
 		})
 		.then(() => {
-			eventTimer.trigger('slotReady', adName);
```
[the remark this answers is not in the document]
Comment (Jakeii) [reply bundle/src/projects/commercial/modules/dfp/load-advert.ts:49]: we do, it's measuring `advert.whenSlotReady` I think? not sure why it's not in there tbh

File: bundle/src/projects/common/modules/experiments/tests/eager-prebid.ts (hunk 1 above)
Comment (Jakeii) [reply bundle/src/projects/common/modules/experiments/tests/eager-prebid.ts:17]: I was considoring that, but I've already given them to Zoyah now🙈

**Comment by mchv** [comment]
This is excellent work @Jakeii 💯

**Comment by Jakeii** [comment]
Thanks @mchv!

**Pull request merged**